### PR TITLE
[PagerDuty] Add more code examples

### PR DESCRIPTION
**Pull request by ls339**
### What does this PR do?

Update the API documentation to add more code examples for different operations.

### Motivation
Some customers have mistakenly updated their configuration with a bad value. Adding more examples for append (POST) operations.

### Preview link
https://docs.datadoghq.com/api/?lang=bash#pagerduty

### CODE REVIEW

File: content/api/integrations/pagerduty.md [hunk 1]
```diff
@@ -7,33 +7,34 @@ external_redirect: /api/#pagerduty
...
 ##### ARGUMENTS
 
-* **`services`** [*required*]:    
-    Array of PagerDuty service objects. [Learn how to configure you Datadog service with PagerDuty documentation][2]. A PagerDuty service object is composed by:  
-    
-    * **`service_name`** [*required*]:  
+* **`services`** :
+    Array of PagerDuty service objects. [Learn how to configure you Datadog service with PagerDuty documentation][2]. A PagerDuty service object is composed by:
+
+    * **`service_name`** :
```
[the remark this answers is not in the document]
Comment (ls339) [reply content/api/integrations/pagerduty.md]: Yes, thats correct. While `services` is optional, the elements of the array are not. 👍

File: content/api/integrations/code_snippets/api-integrations-pagerduty.sh [hunk 2]
```diff
@@ -16,6 +17,38 @@ curl -v -X PUT -H "Content-type: application/json" \
...
+-d '{
+      "subdomain": "<PAGERDUTY_SUB_DOMAIN>",
+      "schedules": ["<SCHEDULE_1>", "<SCHEDULE_2>"],
+      "api_token": "<PAGERDUTY_TOKEN>"
+  }' \
+"https://api.datadoghq.com/api/v1/integration/pagerduty?api_key=${api_key}&application_key=${app_key}"
 
-curl -v -X DELETE "https://api.datadoghq.com/api/v1/integration/pagerduty?api_key=${api_key}&application_key=${app_key}"
\ No newline at end of file
+# Append to the `services` and `schedules` list.
```
Comment (yannmh) [inline content/api/integrations/code_snippets/api-integrations-pagerduty.sh]: "Add new services and schedules to the existing configuration" instead? thoughts?

File: content/api/integrations/pagerduty.md [hunk 3]
```diff
@@ -7,33 +7,34 @@ external_redirect: /api/#pagerduty
...
+Configure your Datadog-PagerDuty integration directly through the Datadog API.
 [Read more about Datadog-PagerDuty integration][1]
 
-**Note**: 
+**Note**:
 
-* Using the `POST` method updates your integration configuration by **adding** your new configuration to the existing one in your Datadog organization. 
+* Using the `POST` method updates your integration configuration by **adding** your new configuration to the existing one in your Datadog organization.
 * Using the `PUT` method updates your integration configuration by **replacing** your current configuration with the new one sent to your Datadog organization.
+* All arguments are required in the payload **only** when creating (`PUT`) a new PagerDuty configuration.
```
Comment (yannmh) [inline content/api/integrations/pagerduty.md]: The wording "in the payload **only**" is confusing.  Specifically the words "payload" and "only". I'd just remove both, what do you think @cswatt ?

Comment (ls339) [reply content/api/integrations/pagerduty.md]: 👍

File: content/api/integrations/pagerduty.md [hunk 4]
```diff
@@ -27,14 +27,14 @@ Configure your Datadog-PagerDuty integration directly through the Datadog API.
...
+* **`subdomain`** [*optional*, *default*=**None**]:
     Your PagerDuty account's personalized subdomain name.
 
-* **`schedules`** :
+* **`schedules`** [*required*]::
     Array of your schedule URLs, e.g:
      `["https://my-pd.pagerduty.com/schedules#PCPYT4M", "https://my-pd.pagerduty.com/schedules#PKTPB7P"]`
 
-* **`api_token`** :
+* **`api_token`** [*required*]:
```
Comment (ls339) [inline content/api/integrations/pagerduty.md]: @l0k0ms I feel that tagging these arguments as `required` is a bit misleading. Its only required when creating a new configuration. This has caused customers to mistakenly update values in an existing configuration. My solution was to add a note `All arguments are required when creating (`PUT`) a new PagerDuty configuration` and removing `required`. What are your thoughts on this? How can we make this clearer?

Comment (l0k0ms) [reply content/api/integrations/pagerduty.md]: 🤔 I missed that, it might be the best way to go indeed, let me update the PR

**Pull request merged**